colossalai/nn/optimizer/lamb.py

```diff
@@ -79,11 +81,15 @@ def step(self, closure=None):
                 # v_t
                 exp_avg_sq.mul_(beta2).addcmul_(grad, grad, value=1 - beta2)
 
-                # Paper v3 does not use debiasing.
-                # bias_correction1 = 1 - beta1 ** state['step']
-                # bias_correction2 = 1 - beta2 ** state['step']
-                # Apply bias to lr to avoid broadcast.
-                # * math.sqrt(bias_correction2) / bias_correction1
+                # NOTE: Paper v3 does not use debiasing.
+                if group["bias_correction"]:
+                    bias_correction1 = 1 - beta1 ** state["step"]
+                    bias_correction2 = 1 - beta2 ** state["step"]
+                    # Apply debiasing to lr to avoid broadcast
+                    group["lr"] *= (bias_correction2**0.5) / bias_correction1
```

Comment:
We should not modify lr directly.

Reply:
avoid broadcast?

Reply:
This is to avoid broadcast. It's also mathematically equivalent.

Reply:
This breaks lr scheduler. Use `scaled_lr *= (bias_correction2**0.5) / bias_correction1`